src/main/java/com/arpnetworking/metrics/mad/model/json/Version2g.java

```diff
@@ -0,0 +1,405 @@
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import net.sf.oval.constraint.MatchPattern;
+import net.sf.oval.constraint.NotEmpty;
+import net.sf.oval.constraint.NotNull;
+import org.joda.time.DateTime;
+
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Model for the version 2g query log line.
+ *
+ * Notes:
+ * <ul>
+ * <li>For optional fields null and unspecified are treated the same. Where
+ * appropriate this means null is mapped to a default value (e.g. empty map) or
+ * wrapped in an <code>Optional</code>.</li>
+ * </ul>
+ *
+ * @author Ryan Ascheman (rascheman at groupon dot com)
+ */
+public final class Version2g {
+
+    public Map<String, Element> getTimers() {
+        return _timers;
+    }
+
+    public Map<String, Element> getGauges() {
+        return _gauges;
+    }
+
+    public Map<String, Element> getCounters() {
+        return _counters;
+    }
+
+    public Annotations getAnnotations() {
+        return _annotations;
+    }
+
+    public Map<String, String> getDimensions() {
+        return _dimensions;
+    }
+
+    public String getVersion() {
+        return _version;
+    }
+
+    private Version2g(final Builder builder) {
+        _dimensions = builder._dimensions == null ? ImmutableMap.of() : ImmutableMap.copyOf(builder._dimensions);
+        _annotations = builder._annotations;
+        _version = builder._version;
+        _timers = builder._timers == null ? ImmutableMap.of() : ImmutableMap.copyOf(builder._timers);
+        _gauges = builder._gauges == null ? ImmutableMap.of() : ImmutableMap.copyOf(builder._gauges);
+        _counters = builder._counters == null ? ImmutableMap.of() : ImmutableMap.copyOf(builder._counters);
+    }
+
+    private final ImmutableMap<String, String> _dimensions;
+    private final Annotations _annotations;
+    private final ImmutableMap<String, Element> _counters;
+    private final ImmutableMap<String, Element> _timers;
+    private final ImmutableMap<String, Element> _gauges;
+    private final String _version;
+
+    /**
+     * Builder for the Data class.
+     */
+    public static final class Builder extends OvalBuilder<Version2g> {
+        /**
+         * Public constructor.
+         */
+        public Builder() {
+            super(Version2g::new);
+        }
+
+        /**
+         * Sets the annotations field.
+         *
+         * @param value Value
+         * @return This builder
+         */
+        public Builder setAnnotations(final Annotations value) {
+            _annotations = value;
+            return this;
+        }
+
+        /**
+         * Sets the dimensions field.
+         *
+         * @param value Value
+         * @return This builder
+         */
+        public Builder setDimensions(final Map<String, String> value) {
```

Comment:
This should probably match the Annotations strategy, i.e. using a new `Dimensions` class.